### PR TITLE
feat: new metering point connection state validation rule

### PR DESCRIPTION
**Pull request by MadsDue**
## Description

When receiving metered data, the metering point connection state must be connected or disconnected. Otherwise, we should reject submitting metered data for the metering point.

## References

Link to assignment:

## Checklist

- [ ] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [ ] Has it been considered whether data is being delivered to the wrong actor?
- [x] Subsystem test executed (dev_002/dev_003) => https://github.com/Energinet-DataHub/dh3-environments/actions/runs/14061751928
- [x] Is there time to monitor state of the release to Production?
- [x] Reference to the task => https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/434

### CODE REVIEW

**Review by MWO1024 — APPROVED** [review]

File: source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/V1/BusinessValidation/ConnectionStateValidationRuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+using FluentAssertions;
+using Microsoft.FeatureManagement;
+using Moq;
+using NodaTime;
+
+namespace Energinet.DataHub.ProcessManager.Orchestrations.Tests.Unit.Processes.BRS_021.ForwardMeteredData.V1.BusinessValidation;
+
+public class ConnectionStateValidationRuleTests
+{
+    private readonly ConnectionStateValidationRule _sut = new();
```
Comment (MWO1024) [inline source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/V1/BusinessValidation/ConnectionStateValidationRuleTests.cs:31]: Before I forget: We should add a test case with the empty set of master data. The validation should ideally not fail (otherwise it may block deployments). And the case for the empty set of master data is handled by another validation.

File: source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/V1/BusinessValidation/ConnectionStateValidationRuleTests.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                [],
+                connectionState,
+                MeteringPointType.Production,
+                MeteringPointSubType.Physical,
+                Resolution.Hourly,
+                MeasurementUnit.KilowattHour,
+                "product",
+                null,
+                ActorNumber.Create("1111111111112")),
+        ]));
```
Comment (MWO1024) [inline source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/V1/BusinessValidation/ConnectionStateValidationRuleTests.cs:60]: Consider using multiple (two) master data chunks with only one faulty to ensure we correctly trigger the validation error as long as at least one of them are wrong and not accidentally only when they are all wrong.

**Review by MWO1024 — APPROVED** [review]

**Review by MWO1024 — APPROVED** [review]
